dsl/pact.go

```diff
@@ -126,9 +142,13 @@ func (p *Pact) Setup(startMockServer bool) *Pact {
 			p.Consumer,
 			"--provider",
 			p.Provider,
+			"--port",
+			fmt.Sprintf("%d", port),
 		}
 
 		p.Server = p.pactClient.StartServer(args)
+		// StartServer doesn't know the port yet, so we need to update the server here
+		p.Server.Port = port
```

Comment:
Might need to pass por tto `StartServer` since mockclient ends up calling `waitFor` on the port. (current reason why tests fail)